### PR TITLE
cpputest: fixed wrap

### PR DESCRIPTION
**Pull request by SteelPh0enix**
Current wrap of CppUTest is fundamentally broken in multiple ways:

* Overwrites main dependency (by default), making it impossible to link it and use the wrap
* It's ignoring the existence of platform-specific implementations, forcing the user to provide them manually
* Typo in options

This PR fixes all those issues and makes this wrap usable.

### CODE REVIEW

**Comment by SteelPh0enix** [comment]
Any suggestions how could i keep the extensions off by default while keeping the CI green?

**Comment by jpakkane** [comment]
You can specify build options in `ci_config.json`.

**Comment by SteelPh0enix** [comment]
I've added CI settings, fixed some stuff and performed sanity checks locally, it should be green now.

**Comment by SteelPh0enix** [comment]
Huh, so that's how this switch works...

All right, i'll make a compiler check in there for MSVC and GCC/Clang, should be good enough for the CI and "most people".

**Comment by SteelPh0enix** [comment]
Done, hopefully that's the last issue

File: subprojects/packagefiles/cpputest/src/Platforms/meson.build [hunk 1]
```diff
@@ -3,7 +3,7 @@ cpputest_platform = get_option('cpputest_platform')
 if cpputest_platform != 'none'
   if cpputest_platform == 'autodetect'
     compiler_id = meson.get_compiler('cpp').get_id()
-    if compiler_id == 'clang' or compiler_id == 'gcc'
+    if compiler_id == 'gcc' or compiler_id == 'clang' or compiler_id == 'clang-cl'
```
Comment (neheb) [inline subprojects/packagefiles/cpputest/src/Platforms/meson.build:6]: should it be treated as GCC or VisualCpp?

**Comment by SteelPh0enix** [comment]
Hm, maybe `clang-cl` should be bound to `VisualCpp` instead...
Yet the CI passes, hopefully it stays that way. Leaving that comment for the future me if this ever becomes a problem. It's 2AM, CI is green, I'm going to sleep.

File: subprojects/packagefiles/cpputest/src/Platforms/meson.build (hunk 1 above)
Comment (SteelPh0enix) [reply subprojects/packagefiles/cpputest/src/Platforms/meson.build:6]: Yeah, it passes here so I'm not sure it's just a funky compatibility layer, or intended behaviour.
I'll throw it to VisualCpp and let's see what happens

Comment (SteelPh0enix) [reply subprojects/packagefiles/cpputest/src/Platforms/meson.build]: After quick read, yeah, this should definitely be treated as VisualCpp

**Pull request merged**